libvips/colour/icc_transform.c

```diff
@@ -545,6 +545,54 @@ vips_icc_print_profile( const char *name, cmsHPROFILE profile )
 }
 #endif /*DEBUG*/
 
+/* How many bands we expect to see from an image after preprocessing by our
+ * parent classes. This is a bit fragile :-( 
+ *
+ * FIXME ... split the _build() for colour into separate preprocess / process
+ * / postprocess phases so we can load profiles after preprocess but before
+ * actual processing takes place.
+ */
+static int
+vips_image_expected_bands( VipsImage *image )
+{
+	int expected_bands;
+
+	switch( image->Type ) {
+		case VIPS_INTERPRETATION_B_W:
+		case VIPS_INTERPRETATION_GREY16:
+			expected_bands = 1;
+			break;
+
+		case VIPS_INTERPRETATION_XYZ:
+		case VIPS_INTERPRETATION_LAB:
+		case VIPS_INTERPRETATION_LABQ:
+		case VIPS_INTERPRETATION_RGB:
+		case VIPS_INTERPRETATION_CMC:
+		case VIPS_INTERPRETATION_LCH:
+		case VIPS_INTERPRETATION_LABS:
+		case VIPS_INTERPRETATION_sRGB:
+		case VIPS_INTERPRETATION_YXY:
+		case VIPS_INTERPRETATION_RGB16:
+		case VIPS_INTERPRETATION_scRGB:
+		case VIPS_INTERPRETATION_HSV:
+			expected_bands = 3;
+			break;
+
+		case VIPS_INTERPRETATION_MULTIBAND:
+		case VIPS_INTERPRETATION_HISTOGRAM:
+		case VIPS_INTERPRETATION_CMYK:
```

Comment:
Note: previously images tagged as CMYK had `expected_bands` set to `4`, but this would break NCLR support.